### ISSUE TITLE
Alternating tags in volume

### ISSUE THREAD

**Issue by heldersepu** [hashicorp/terraform-provider-aws#5080]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

```
Terraform v0.11.7
+ provider.aws v1.23.0
+ provider.template v1.0.0
```

### Affected Resource(s)

* aws_instance
* aws_ebs_volume
* aws_volume_attachment

### Debug Output
- First apply we get:
```volume_tags.Name: "suse12 volume tags" => "suse12-data-ndj"```
- Second apply we get:
```volume_tags.Name: "suse12-data-ndj" => "suse12 volume tags"```

Here is a longer output
```
PS C:\Users\helde\Documents\GitHub\hs-scripts\TerraForm\suse> terraform apply
aws_ebs_volume.data-ndj: Refreshing state... (ID: vol-0e4ee5ad2c9a3629e)
data.aws_ami.suse12: Refreshing state...
aws_instance.suse12: Refreshing state... (ID: i-0140a00017d5c96dc)
aws_volume_attachment.data-ndj: Refreshing state... (ID: vai-2788403704)

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  ~ aws_ebs_volume.data-ndj
      tags.Name: "suse12 volume tags" => "suse12-data-ndj"


Plan: 0 to add, 1 to change, 0 to destroy.
...

PS C:\Users\helde\Documents\GitHub\hs-scripts\TerraForm\suse> terraform apply
aws_ebs_volume.data-ndj: Refreshing state... (ID: vol-0e4ee5ad2c9a3629e)
data.aws_ami.suse12: Refreshing state...
aws_instance.suse12: Refreshing state... (ID: i-0140a00017d5c96dc)
aws_volume_attachment.data-ndj: Refreshing state... (ID: vai-2788403704)

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  ~ aws_instance.suse12
      volume_tags.Name: "suse12-data-ndj" => "suse12 volume tags"


Plan: 0 to add, 1 to change, 0 to destroy.
...
```

### Expected Behavior
The Volume tag set at `aws_ebs_volume` should not be overwritten by those on `volume_tags`

### Actual Behavior
On every apply Tags are alternating 

### Steps to Reproduce

terraform apply this:
https://github.com/heldersepu/hs-scripts/blob/master/TerraForm/suse/ec2.tf

**Comment by wblakecannon** [comment]
Also having this issue with v0.12.5. Looking forward to a fix.

**Comment by YakDriver** [comment]
We have merged a fix to the `volume_tags` issue in #15474. We have added tests to cover the issues observed. Please note that using `volume_tags` in `aws_instance` is _not_ compatible with using `tags` in `aws_ebs_volume`. You need to use one or the other. Prior to this fix, even following this rule, you would encounter errors. Along with the fix, we've added `tags` to the `root_block_device` and `ebs_block_device` configuration blocks in `aws_instance`.

Now that the fix is in place, if you find any problems with `volume_tags`, let us know by [opening a new issue](https://github.com/hashicorp/terraform-provider-aws/issues/new/choose).
